analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpExecutableLinesMetric.cs

```diff
@@ -25,34 +25,30 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Extensions;
 using SonarAnalyzer.Helpers;
 using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Metrics.CSharp
 {
     public static class CSharpExecutableLinesMetric
     {
-        private const int AttributeSuffixIndex = 9;
-
         public static ImmutableArray<int> GetLineNumbers(SyntaxTree syntaxTree, SemanticModel semanticModel)
         {
             var walker = new ExecutableLinesWalker(semanticModel);
             walker.SafeVisit(syntaxTree.GetRoot());
-
             return walker.ExecutableLineNumbers.ToImmutableArray();
         }
 
         private sealed class ExecutableLinesWalker : SafeCSharpSyntaxWalker
         {
-            private readonly SemanticModel semanticModel;
-
-            public ExecutableLinesWalker(SemanticModel semanticModel)
-            {
-                this.semanticModel = semanticModel;
-            }
+            private readonly SemanticModel model;
 
             public HashSet<int> ExecutableLineNumbers { get; } = new();
 
+            public ExecutableLinesWalker(SemanticModel model) =>
+                this.model = model;
+
             public override void DefaultVisit(SyntaxNode node)
```

Comment:
Yes, it's doable with `DescendantNodes` too. And there would have to be some side effects of the `descentIntoChildren`, as it would need to feed the `ExecutableLineNumbers` on the fly. 

So this visitor misses the original intention of the class and provides a clean design at the same time.